x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/use_create_analytics_form.ts

```diff
@@ -94,6 +95,10 @@ export const useCreateAnalyticsForm = (): CreateAnalyticsFormProps => {
       ? jobConfig
       : getJobConfigFromFormState(form);
 
+    if (isAdvancedEditorEnabled) {
+      destinationIndex = analyticsJobConfig.dest?.index as string;
```

Comment:
Added the string type here because `analyticsJobConfig` has `DeepPartial` type so index is being seen as type string or undefined though it will always be defined here since the config has to be valid to get to this creation point.